data_sources/amanda_closure_publishing.py

```diff
@@ -0,0 +1,234 @@
+        elif permit_type == "EX":
+            if subtype == 50685:
+                description = f"Excavation Permit has been issued for this location."
+                name = "WorkZone Event"
+            else:
+                description = f"Excavation Permit has been issued for this location. \n Details: {folderdesc}"
+                name = foldername
+
+            data_source_id = amanda_ex_id
+
+        segments = closures["SEGMENT_ID"].unique()
+
+        start_date = closures["START_DATE"].iloc[0]
+        end_date = closures["END_DATE"].iloc[0]
+
+        # checking if the closure is some time in the future.
+        # adding one hour to the end time to help inform consumers that the work zone has officially ended.
+        if end_date + datetime.timedelta(hours=1) > current_time:
+            wz = AmandaWorkZone(
+                data_source_id=data_source_id,
+                name=name,
+                folderrsn=p,
+                description=description,
+                start_date=start_date.tz_convert("UTC").strftime("%Y-%m-%dT%H:%M:%SZ"),
+                end_date=end_date.tz_convert("UTC").strftime("%Y-%m-%dT%H:%M:%SZ"),
+            )
+            for s in segments:
+                seg = closures[closures["SEGMENT_ID"] == s]
+                if "Closure : Full Road" in list(seg["CLOSURE_TYPE"]):
+                    if s in segment_lookup:
+                        wz.add_closure(s, "all-lanes-closed", segment_lookup[s])
+                    else:
+                        logger.info(
+                            f"{s} not found in street segments feature layer under folderrsn {p}"
+                        )
+                elif "Traffic Lane : Dimensions" in list(
+                    seg["CLOSURE_TYPE"]
+                ) or "Open Cuts : Street" in list(seg["CLOSURE_TYPE"]):
+                    if s in segment_lookup:
+                        wz.add_closure(s, "some-lanes-closed", segment_lookup[s])
+                    else:
+                        logger.info(
+                            f"{s} not found in street segments feature layer under folderrsn {p}"
+                        )
+            if wz.get_number_of_closures() > 0:
+                work_zones.append(wz)
+
+    feed_info = create_feed_info(amanda_turp_id, amanda_ex_id, current_time)
+
+    features = []
+    for wz in work_zones:
+        wz.reduce_closure_geometry()
+        features += wz.generate_json()
+
+    output = {"feed_info": feed_info, "type": "FeatureCollection", "features": features}
+
+    if SO_USER and SO_PASS:
+        logger.info("Uploading data to Socrata")
+        # logging in with sodapy
+        soda = Socrata(
+            SO_WEB,
+            SO_TOKEN,
+            username=SO_USER,
+            password=SO_PASS,
+            timeout=500,
+        )
+
+        files = {"file": ("wzdx_atx.geojson", json.dumps(output))}
+        response = soda.replace_non_data_file(FEED_DATASET, {}, files)
+        logger.info(response)
+
```

Comment:
what do you think about adding another logging line here, to say that the flat dataset is next? Just thinking about debugging this in the future....if theres an error in the response, remembering which one came first.